sdk/core/Azure.Core/src/Pipeline/Internal/RetryPolicy.cs

```diff
@@ -212,4 +233,28 @@ private TimeSpan CalculateExponentialDelay(int attempted)
                     _maxDelay.TotalMilliseconds));
         }
     }
+
+#pragma warning disable SA1402 // File may only contain a single type
+    internal class MaxAttemptsRetryCondition : RetryCondition
```

Comment:
We could also make this public if we wanted to provide a library of possible retry conditions.

Reply:
What is the benefit of this as opposed to the RetryOptions setting?

Reply:
I don't understand the question ... which RetryOptions setting?

Reply:
Ah, maybe I understand ... this works with the RetryOptions list of RetryConditions ... it's a concrete implementation of a RetryCondition, which we could use by default to maintain the existing RetryPolicy behavior.

Reply:
So if retry conditions are specified, the other retry options are ignored?

Reply:
No, everything in RetryOptions is applied if you use the internal RetryPolicy.

Reply:
Ah, I see. This is how we implement the MaxAttempts logic.